lib/core/src/send_swap.rs

```diff
@@ -150,25 +155,45 @@ impl SendSwapStateHandler {
                 | SubSwapStates::SwapExpired,
             ) => {
                 match swap.lockup_tx_id {
-                    Some(_) => match swap.refund_tx_id {
-                        Some(refund_tx_id) => warn!(
+                    Some(_) => {
+                        match swap.refund_tx_id {
+                            Some(refund_tx_id) => warn!(
                         "Refund tx for Send Swap {id} was already broadcast: txid {refund_tx_id}"
                     ),
-                        None => {
-                            warn!("Send Swap {id} is in an unrecoverable state: {swap_state:?}, and lockup tx has been broadcast. Attempting refund.");
-
-                            let refund_tx_id = self.refund(&swap).await?;
-                            info!("Broadcast refund tx for Send Swap {id}. Tx id: {refund_tx_id}");
-                            self.update_swap_info(
-                                id,
-                                RefundPending,
-                                None,
-                                None,
-                                Some(&refund_tx_id),
-                            )
-                            .await?;
+                            None => {
+                                warn!("Send Swap {id} is in an unrecoverable state: {swap_state:?}, and lockup tx has been broadcast. Attempting refund.");
+
+                                let mut refund_attempts = 0;
+                                while refund_attempts < MAX_REFUND_ATTEMPTS {
```

Comment:
Sorry for responding late but won't this block the task that handles swap updates?